{"uid":"6379b16dfcf419bc","name":"test_change_country_invalid_data[invalid_payload2-authorize_admin]","fullName":"tests.orders.test_group_order_change_country#test_change_country_invalid_data","historyId":"aec72a090d429759b7ea9a1a03a67739","time":{"start":1744504692574,"stop":1744504693009,"duration":435},"description":"Негативные тесты: передача некорректных данных","descriptionHtml":"<p>Негативные тесты: передача некорректных данных</p>\n","status":"skipped","statusMessage":"XFAIL https://ru.yougile.com/team/01c2a92f1328/#IT-507\n\nAssertionError: Ожидался статус 400, но получен 405","statusTrace":"api_client = <api_tests.utils.api_client.APIClient object at 0x7fd6819b3250>\nauthorize_func = <function authorize_admin at 0x7fd681e72520>\ninvalid_payload = {'country_id': None, 'int_id': 1}\n\n    @pytest.mark.xfail(reason = \"https://ru.yougile.com/team/01c2a92f1328/#IT-507\")\n    @pytest.mark.parametrize(\"authorize_func\", [authorize_client, authorize_partner, authorize_admin])\n    @pytest.mark.parametrize(\"invalid_payload\", [\n        {},  # Пустой запрос\n        {\"int_id\": None, \"country_id\": 1},  # Нет ID заказа\n        {\"int_id\": 1, \"country_id\": None},  # Нет ID страны\n    ])\n    def test_change_country_invalid_data(api_client, authorize_func, invalid_payload):\n        \"\"\"Негативные тесты: передача некорректных данных\"\"\"\n        authorize_func(api_client)\n        response = api_client.post(\"/orders/group_order/change_country\", json=invalid_payload)\n>       assert_response_status_and_error(response, 400)\n\napi_tests/tests/orders/test_group_order_change_country.py:35: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nresponse = {'error': '', 'status_code': 405}, expected_status = 400\n\n    def assert_response_status_and_error(response, expected_status):\n        actual_status = response[\"status_code\"]\n>       assert actual_status == expected_status, f\"Ожидался статус {expected_status}, но получен {actual_status}\"\nE       AssertionError: Ожидался статус 400, но получен 405\n\napi_tests/utils/asserts.py:39: AssertionError","flaky":false,"newFailed":false,"newBroken":false,"newPassed":false,"retriesCount":0,"retriesStatusChange":false,"beforeStages":[{"name":"api_client","time":{"start":1744504686358,"stop":1744504686358,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false}],"afterStages":[{"name":"api_client::0","time":{"start":1744504695225,"stop":1744504695225,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false}],"labels":[{"name":"tag","value":"@pytest.mark.xfail(reason='https://ru.yougile.com/team/01c2a92f1328/#IT-507')"},{"name":"parentSuite","value":"tests.orders"},{"name":"suite","value":"test_group_order_change_country"},{"name":"host","value":"fv-az1706-77"},{"name":"thread","value":"1832-MainThread"},{"name":"framework","value":"pytest"},{"name":"language","value":"cpython3"},{"name":"package","value":"tests.orders.test_group_order_change_country"},{"name":"resultFormat","value":"allure2"}],"parameters":[{"name":"authorize_func","value":"<function authorize_admin at 0x7fd681e72520>"},{"name":"invalid_payload","value":"{'int_id': 1, 'country_id': None}"}],"links":[],"hidden":false,"retry":false,"extra":{"severity":"normal","retries":[],"categories":[],"tags":["@pytest.mark.xfail(reason='https://ru.yougile.com/team/01c2a92f1328/#IT-507')"]},"source":"6379b16dfcf419bc.json","parameterValues":["<function authorize_admin at 0x7fd681e72520>","{'int_id': 1, 'country_id': None}"]}